tests/phpunit/tests/admin/includesPlugin.php

```diff
@@ -257,7 +257,13 @@ public function data_submenu_position() {
 			array( -1, 0 ),                    // Negative numbers are treated the same as passing 0.
 			array( -7, 0 ),                    // Negative numbers are treated the same as passing 0.
 			array( 1, 1 ),                     // Insert as the second item.
+			array( '1', 1 ),                   // Insert as the second item.
+			array( 1.5, 1 ),                   // Insert as the second item.
+			array( '1.5', 1 ),                 // Insert as the second item.
 			array( 3, 3 ),                     // Insert as the 4th item.
+			array( '3', 3 ),                   // Insert as the 4th item.
+			array( 3.5, 3 ),                   // Insert as the 4th item.
+			array( '3.5', 3 ),                 // Insert as the 4th item.
```

Comment:
Also try testing for `(string) "42e0"` and other values that return `true` in `is_numeric()`.

Reply:
> Also try testing for `(string) "42e0"` and other values that return `true` in `is_numeric()`.

Good call, added `3e0` in e0ceb288b4f4d4f472d0a260f0c06e8c3ace3f08

Unfortunately PHP broke backcompat for items some with whitespace in 8.0 (eg ` 100 `) so I've left them alone rather than make the tests conditional for an edge case.

Reply:
The white space was helpfully trimmed in my example `[space]100[space]`.